src/compositemark/boxplot.ts

```diff
@@ -48,71 +48,44 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT
   const {color: _color, ...nonPositionEncodingWithoutColorSize} = nonPositionEncodingWithoutSize;
   const midTickAndBarSizeChannelDef = size ? {size: size} : {size: {value: config.box.size}};
 
-  let discreteAxisFieldDef;
+  let kIQRScalar: number = undefined;
+  if (isBoxPlotDef(mark)) {
+    if (mark && mark.extent) {
+      if(typeof(mark.extent) === 'number') {
+        kIQRScalar = mark.extent;
+      }
+    }
+  }
+  const isMinMax = kIQRScalar === undefined;
+
+  let discreteAxisFieldDef: PositionFieldDef<Field>;
   let continuousAxisChannelDef: PositionFieldDef<Field>;
   let discreteAxis;
   let continuousAxis;
 
-  if (isFieldDef(encoding.x) && isFieldDef(encoding.y)) {
-    // 2D
-
-    const orient: Orient = box2DOrient(spec);
-    const params = box2DParams(spec, orient);
-    discreteAxisFieldDef = params.discreteAxisFieldDef;
-    continuousAxisChannelDef = params.continuousAxisChannelDef;
-    discreteAxis = params.discreteAxis;
-    continuousAxis = params.continuousAxis;
-
-  } else if (isFieldDef(encoding.x) && isContinuous(encoding.x) && encoding.y === undefined) {
-    // 1D horizontal
-    continuousAxis = 'x';
-    continuousAxisChannelDef = encoding.x;
-  } else if (encoding.x === undefined && isFieldDef(encoding.y) && isContinuous(encoding.y)) {
-    // 1D vertical
-    continuousAxis = 'y';
-    continuousAxisChannelDef = encoding.y;
-  } else {
-    throw new Error('Need a continuous axis for 1D boxplots');
+  if (encoding.x === undefined && encoding.y === undefined) {
+    throw new Error('Need at least one axis');
   }
 
+  const orient: Orient = boxOrient(spec);
+  const params = boxParams(spec, orient);
+  discreteAxisFieldDef = params.discreteAxisFieldDef;
+  continuousAxisChannelDef = params.continuousAxisChannelDef;
```

Comment:
Why is one of them `...ChannelDef`, but another is `...FieldDef`?  
Let's use "ChannelDef" for both?